core/domain/topic_services.py

```diff
@@ -914,6 +923,24 @@ def publish_topic(topic_id, committer_id):
     })]
     save_topic_rights(
         topic_rights, committer_id, 'Published the topic', commit_cmds)
+    _create_exploration_opportunities_for_topic(topic)
+
+
+def _create_exploration_opportunities_for_topic(topic):
```

Comment:
Similar to my reasoning above, I wanted to avoid multiple topic/story get calls since in the places where the opportunities are created, the topic or story have already been fetched.